libs/utils/src/http/endpoint.rs

```diff
@@ -27,14 +28,38 @@ static SERVE_METRICS_COUNT: Lazy<IntCounter> = Lazy::new(|| {
     .expect("failed to define a metric")
 });
 
+static X_REQUEST_ID_HEADER_STR: &str = "x-request-id";
+
+static X_REQUEST_ID_HEADER: HeaderName = HeaderName::from_static(X_REQUEST_ID_HEADER_STR);
+#[derive(Debug, Default, Clone)]
+struct RequestId {
```

Comment:
1. Box<str> should be a slightly better option (a bit less size since its immutable)
2. There are several options how to declare that. You can use newtype: 

```rust
struct RequestId(Box<str>)
```
Though .id will become .0

Completely optional, up to you